### PR TITLE
issue: 1079098 vma_extra fix array out of range

### PR DESCRIPTION
**Pull request by rafiw**
Signed-off-by: Rafi Wiener <rafiw@mellanox.com>

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1479/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1484/ for details (Mellanox internal link).

File: src/vma/sock/sock-redirect.cpp [hunk 1]
```diff
@@ -508,14 +508,14 @@ int vma_get_socket_rings_fds(int fd, int *ring_fds, int ring_fds_sz)
 	if (ring_fds_sz > 0) {
```
Comment (ilansmith) [inline src/vma/sock/sock-redirect.cpp]: I would replace this with:

```
if (ring_fds_sz <= 0) {
        errno = EINVAL;
        return -1;
}
```

**Comment by ilansmith** [comment]
:+1:

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1492/ for details (Mellanox internal link).

File: src/vma/sock/sock-redirect.cpp [hunk 2]
```diff
@@ -504,18 +504,20 @@ int vma_get_socket_rings_fds(int fd, int *ring_fds, int ring_fds_sz)
...
-		if (p_socket_object && p_socket_object->check_rings()) {
-				p_rings_fds = p_socket_object->get_rings_fds(rings_num);
-				for (int i = 0; i < rings_num; i++) {
-					ring_fds[i] = p_rings_fds[i];
-				}
+	
+	if (ring_fds_sz <= 0) {
+        	errno = EINVAL;
+        	return -1;
+	}
```
Comment (igor-ivanov) [inline src/vma/sock/sock-redirect.cpp]: please consider checking `ring_fds != NULL` as well

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1518/ for details (Mellanox internal link).

**Comment by rafiw** [comment]
bot:retest

File: src/vma/sock/sock-redirect.cpp (hunk 2 above)
Comment (rafiw) [inline src/vma/sock/sock-redirect.cpp]: done

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1520/ for details (Mellanox internal link).

**Comment by ilansmith** [comment]
Merged #432

**Pull request closed without merging**